bigtable/tests/admin_integration_test.cc

```diff
@@ -12,185 +12,280 @@
+            bigtable::testing::TableTestEnvironment::project_id(),
+            bigtable::ClientOptions());
+    table_admin_ = absl::make_unique<bigtable::TableAdmin>(
+        admin_client, bigtable::testing::TableTestEnvironment::instance_id());
+  }
+
+  void TearDown() {}
+
+  bool TestForTableListCheck(std::vector<std::string> expected_table_list) {
+    std::vector<std::string> actual_table_list;
+    std::vector<std::string> diff_table_list;
+
+    auto table_list = table_admin_->ListTables(admin_proto::Table::NAME_ONLY);
+    for (auto const& table_name : table_list) {
+      actual_table_list.push_back(table_name.name());
+    }
+
+    // Sort actual_table_list
+    sort(actual_table_list.begin(), actual_table_list.end());
+    // Sort expected_table_list
+    sort(expected_table_list.begin(), expected_table_list.end());
+
+    // Get the difference of expected_table_list and actual_table_list
+    set_difference(expected_table_list.begin(), expected_table_list.end(),
+                   actual_table_list.begin(), actual_table_list.end(),
+                   back_inserter(diff_table_list));
+
+    if (not diff_table_list.empty()) {
+      std::ostringstream ostring_stream;
+      ostring_stream << "Mismatched Tables: "
+                     << absl::StrJoin(diff_table_list, "\n")
+                     << "\nactual: " << absl::StrJoin(actual_table_list, "\n")
+                     << "\nexpected: "
+                     << absl::StrJoin(expected_table_list, "\n") << "\n";
+      std::cout << ostring_stream.str();
+    }
+
+    return diff_table_list.empty();
+  }
+
+  bool CheckTableSchema(admin_proto::Table const& actual_table,
+                        std::string const& expected_text,
+                        std::string const& message) {
+    admin_proto::Table expected_table;
+
+    if (not google::protobuf::TextFormat::ParseFromString(expected_text,
+                                                          &expected_table)) {
+      std::ostringstream ostring_stream;
+      ostring_stream << message << ": could not parse protobuf string <\n"
+                     << expected_text << ">\n";
+      std::cout << ostring_stream.str();
+
+      return false;
+    }
+
+    std::string delta;
+    google::protobuf::util::MessageDifferencer message_differencer;
+    message_differencer.ReportDifferencesToString(&delta);
+    bool is_msg_diff_notpresent =
+        message_differencer.Compare(expected_table, actual_table);
+    if (not is_msg_diff_notpresent) {
+      std::ostringstream ostring_stream;
+      ostring_stream << message << ": mismatch expected vs actual:\n" << delta;
+      std::cout << ostring_stream.str();
+    }
+
+    return is_msg_diff_notpresent;
+  }
+};
+
+}  // anonymus namespace
+
+/**
+ *  Test case for checking if Instance is empty or not.
+ *  If instance is empty then go ahead.
+ *  If instance is not empty the throw runtime error
+ */
+TEST_F(AdminIntegrationTest, CheckInstanceIsEmpty) {
```

Comment:
This probably should be inlined into `main()`.  It is there to provide a good error messages on the program startup if the instance is not suitable for testing.  Right now it runs at some random time and it is not as useful.